### ISSUE TITLE
keyType: "public-key" is superfluous

### ISSUE THREAD

**Issue by balfanz** [w3c/webauthn#554]
I seem to remember that we had discussions about this in the working group and flagged it as something we should come back to later, but we never have.

The `type` attribute in [`PublicKeyCredentialDescriptor`](https://github.com/w3c/webauthn/blob/14c2733ca6a4a9568e4c48fef1b870448818e811/index.bs#L1406) and in [`PublicKeyCredentialParameters`](https://github.com/w3c/webauthn/blob/14c2733ca6a4a9568e4c48fef1b870448818e811/index.bs#L1080) has no meaning and serves no purpose. We should remove it.

**Comment by nadalin** [comment]
@balfanz if no one speaks up then please create a PR and merge it

**Comment by equalsJeffH** [comment]
I disagree.  We need [PublicKeyCredentialType](https://w3c.github.io/webauthn/#credentialType) is an extension point.  we need it to do things like support the millions of UAF-capable mobile handsets over CTAP2.  See: https://github.com/w3c/webauthn/pull/407

See also issue #296

**Comment by rlin1** [comment]
If support renaming it to "assertion format".  Very similar to our extensible attestation statement format.
More specifically I propose:
closing this issue as won't fix and handle the renaming in issue #296.

**Comment by leshi** [comment]
Do UAF authenticators speak CTAP2?

**Comment by leshi** [comment]
When you say "millions of UAF-capable mobile handsets" -- do you mean millions of handsets that have UAF that are going to be upgraded to speak CTAP2?  Or are you saying millions of handsets that could potentially have UAF on them?

If it's future authenticators that don't exist, why won't they be WebAuthn instead of UAF?

**Comment by rlin1** [comment]
The upcoming version of UAF will formally include CTAP2 support (if CTAP2 spec gets finalized soon).
The CTAP2 support is on ASM level that can easily be updated on the handsets.
The Authenticators implemented in the TEE could theoretically be updated, but updating TEEs is more complex is typically not done (other than for fixing severe security issues).

**Comment by leshi** [comment]
Thanks, Rolf!

If an update will be required, why won't the new handsets implement webauthn assertion & attestation format instead?

**Comment by balfanz** [comment]
re/ adding UAF-style assertions to webauthn because we might want to re-use existing UAF keys over CTAP - what would the user experience be for this? Am I imagining this right:

1. User launches PayPal app on phone, which is UAF capable, but before the CTAP update.
2. PayPal app says "hey, let's use fingerprints instead of passwords next time", and uses UAF to accomplish this. Note that the user is still a password user for PayPal, and would use a password to log into PayPal from any other device.
3. Some time later, user launches PayPal app on this phone, does fingerprint, logs in using UAF. Yay! But note that the user is still a password user for PayPal, and would use a password to log into PayPal from any other device.
4. The CTAP update happens for the phone.
5. User goes to paypal.com on their laptop, and types their username ... and their password? (Isn't that what would happen? Why would the user go through the trouble of getting their phone, pairing it with the laptop, waiting for a prompt on the phone, and then use their fingerprint, if they can just log in with their password?)
6. On the laptop, paypal.com says "hey, let's use Windows Hello instead of passwords next time", and uses webauthn (locally on the laptop) to accomplish this.

In other words, I'm not sure why re-using existing UAF keys over CTAP is an important use case. CTAP is for external authenticators, which are used in situations where the user _doesn't_ have a password (or must use the authenticator in addition to the password), which is not what UAF is used for on the millions already-deployed phones.

If, in the future, some of the millions already-deployed phones will be used in a context where they _are_ an external authenticator (e.g., when the user signs up for 2-factor authentication for a service), they can simply create a new webauthn-compatible key pair.

**Comment by rlin1** [comment]
The idea is to use the UAF Authenticator bound the the phone via CTAP2 for the browser on the laptop instead of asking for the password - avoiding the additional identity binding procedure (via password).

**Comment by equalsJeffH** [comment]
@balfanz wrote:
> User goes to paypal.com on their laptop, and types their username ... 
yes (if that is a system whose local storage was cleared or is a system the user has not used before) and then the RP's client-side webapp ought to be able to determine there's a roaming (aka external) authnr in range/connected and give it a try over CTAP2. The existing FIDO (UAF) credential is employed and the user signed-in on the browser on laptop. In the future they can just use their roaming-phone-external authnr with whatever CTAP2-enabled device without necessarily entering their username (modulo RP webapp sophistication).

That said, more generally, i think it would be a mistake to remove the `type` attribute (aka "assertion type" as @rlin1 proposes) in [`PublicKeyCredentialDescriptor`](https://github.com/w3c/webauthn/blob/14c2733ca6a4a9568e4c48fef1b870448818e811/index.bs#L1406) and in [`PublicKeyCredentialParameters`](https://github.com/w3c/webauthn/blob/14c2733ca6a4a9568e4c48fef1b870448818e811/index.bs#L1080) as it is an extension point.

**Comment by nadalin** [comment]
@equalsJeffH  it does not seem like there is a usage for this extension at this time, if there is a proposal please write it up otherwise seems that we close this issue

**Comment by jcjones** [comment]
I agree with closing this, I think it's fine as an extension point.
